<a href="https://colab.research.google.com/github/sidneybraga/Aprendendo-Python/blob/master/udemy2_23.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [4]:
import numpy as np

In [ ]:
from google.colab import files

In [ ]:
uploaded = files.upload()

Saving iris.csv to iris.csv


In [ ]:
base = pd.read_csv('iris.csv')

In [ ]:
base

sepal length  sepal width  petal length  petal width           class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

In [18]:
amostra = np.random.choice(a=[0,1], size = 150, replace=True, p=[0.5, 0.5] )

In [19]:
len(amostra)

150

In [20]:
len(amostra[amostra==1])

59

In [21]:
len(amostra[amostra==0])

91

In [22]:
base['class'].value_counts()

Iris-setosa        50
Iris-virginica     50
Iris-versicolor    50
Name: class, dtype: int64

In [25]:
x, _, y, _ =train_test_split(base.iloc[:,0:4], base.iloc[:,4], test_size=0.5, stratify=base.iloc[:,4])

In [24]:
from sklearn.model_selection import train_test_split

In [26]:
y.value_counts()

Iris-setosa        25
Iris-virginica     25
Iris-versicolor    25
Name: class, dtype: int64

In [27]:
uploaded = files.upload()

Saving infert.csv to infert.csv


In [28]:
infert = pd.read_csv('infert.csv')

In [29]:
infert['education'].value_counts()

6-11yrs    120
12+ yrs    116
0-5yrs      12
Name: education, dtype: int64

In [30]:
(120/148)*100

81.08108108108108

In [31]:
(120/248)*100

48.38709677419355

In [32]:
(116/248)*100

46.774193548387096

In [33]:
(12/248)*100

4.838709677419355

In [34]:
x1, _, y1, _ =train_test_split(infert.iloc[:,2:9], infert.iloc[:,1], test_size=0.6, stratify=infert.iloc[:,1])

In [35]:
y1.value_counts()

6-11yrs    48
12+ yrs    46
0-5yrs      5
Name: education, dtype: int64

In [36]:
from math import ceil

In [37]:
população = 150

In [38]:
amostra1=15

In [39]:
k=população/amostra1

In [59]:
r = np.random.randint(low=1, high=k+1, size=1)

In [47]:
r

array([5])

In [61]:
acumulador=r[0]

In [55]:
sorteados = []

In [63]:
for i in range(amostra1):
  sorteados.append(acumulador)
  acumulador += k

In [64]:
basefinal=base.loc[sorteados]

In [65]:
basefinal

sepal length  sepal width  petal length  petal width            class
4.0             5.0          3.6           1.4          0.2      Iris-setosa
14.0            5.8          4.0           1.2          0.2      Iris-setosa
24.0            4.8          3.4           1.9          0.2      Iris-setosa
34.0            4.9          3.1           1.5          0.1      Iris-setosa
44.0            5.1          3.8           1.9          0.4      Iris-setosa
54.0            6.5          2.8           4.6          1.5  Iris-versicolor
64.0            5.6          2.9           3.6          1.3  Iris-versicolor
74.0            6.4          2.9           4.3          1.3  Iris-versicolor
84.0            5.4          3.0           4.5          1.5  Iris-versicolor
94.0            5.6          2.7           4.2          1.3  Iris-versicolor
104.0           6.5          3.0           5.8          2.2   Iris-virginica
114.0           5.8          2.8           5.1          2.4   Iris-virginica
124.0           6.7          3.3           5.7          2.1   Iris-virginica
134.0           6.1          2.6           5.6          1.4   Iris-virginica
144.0           6.7          3.3           5.7          2.5   Iris-virginica
9.0             4.9          3.1           1.5          0.1      Iris-setosa
19.0            5.1          3.8           1.5          0.3      Iris-setosa
29.0            4.7          3.2           1.6          0.2      Iris-setosa
39.0            5.1          3.4           1.5          0.2      Iris-setosa
49.0            5.0          3.3           1.4          0.2      Iris-setosa
59.0            5.2          2.7           3.9          1.4  Iris-versicolor
69.0            5.6          2.5           3.9          1.1  Iris-versicolor
79.0            5.7          2.6           3.5          1.0  Iris-versicolor
89.0            5.5          2.5           4.0          1.3  Iris-versicolor
99.0            5.7          2.8           4.1          1.3  Iris-versicolor
109.0           7.2          3.6           6.1          2.5   Iris-virginica
119.0           6.0          2.2           5.0          1.5   Iris-virginica
129.0           7.2          3.0           5.8          1.6   Iris-virginica
139.0           6.9          3.1           5.4          2.1   Iris-virginica
149.0           5.9          3.0           5.1          1.8   Iris-virginica

In [8]:
lista= [11.8,12.2,11.8,10.5,13.7]

In [2]:
from scipy import stats

#Biblioteca
Esta bibioteca agrega algumas funções de estatistica(stats)

In [9]:
np.mean(lista)

12.0

In [10]:
np.median(lista)

11.8

In [13]:
quartis = np.quantile(lista, [0, 0.25, 0.5, 0.75, 1])

In [14]:
quartis

array([10.5, 11.8, 11.8, 12.2, 13.7])

In [15]:
np.std(lista)

1.0256705123966463

#Definição de Desvio Padrão:
O desvio padrão é dado pela seguinte equação:
$$ 
\sigma = \sqrt\frac {\sum_{i=1}^{n}(x_i-\bar{x})^2}{n} 
$$
O comando 

```
np.std(lista)
```
retorna o valor do desvio padrão, ou seja:
$$
\sigma= \frac{(11.8-12)^2+(12.2-12)^2+(11.8-12)^2+(10.5-12)^2+(13.7-12)^2}{5} = 1.02
$$ 

O desvio padrão considera que foram coletados todos os dados da população, e é representado pela letra $\sigma$. Este parâmetro nos informa o quão distante estão os dados da população em torno da média, sendo um parâmetro de dispersão; ou seja, quanto maior o valor do desvio padrão, mais dispersos os dados estão uns dos outros.

Enquanto que o desvio padrão amostral é representado pela letra **s** e é dado pela seguinte expressão:

$$ 
s = \sqrt\frac {\sum_{i=1}^{n}(x_i-\bar{x})^2}{n-1} 
$$

Para calcular o desvio padrão amostral, segundo a expressão acima, deve-se utilizar o comando:

```
np.std(lista, ddof=1)

```
cujo o retorno seria dado por:

$$
s= \frac{(11.8-12)^2+(12.2-12)^2+(11.8-12)^2+(10.5-12)^2+(13.7-12)^2}{5-1} = 1.1467
$$ 




In [16]:
np.std(lista, ddof=1)

1.1467344941179711

In [17]:
stats.describe(lista)

DescribeResult(nobs=5, minmax=(10.5, 13.7), mean=12.0, variance=1.314999999999999, skewness=0.2835945083720427, kurtosis=-0.5748890398878097)

#Estatistica ao seu favor:
Os valores da média da mediana podem ser utilizados  a nossa favor no cotidiano, por exemplo, se fossemos ganhar dinhero comprando e vendendo carros, com os dados do mercado, podemos obter a média e a mediana da compra e venda de um determinado modelo. Para ilustrar cnsidere os seguintes valores:

preçoscarros=[35500, 40000, 32000, 43000, 29900]

Para esse exemplo, a média é dado por:


In [18]:
precoscarros=[35500, 40000, 32000, 43000, 29900];

In [19]:
np.mean(precoscarros)

36080.0

E a mediana é dada por:

In [20]:
np.median(precoscarros)

35500.0

Nesse caso, para ser justo e lucrar com o negócio, usaremos o preço da média para venda(36080) e o preço da mediana para compra(35500). 

In [21]:
np.std(precoscarros)

4866.785386679795

In [22]:
np.std(precoscarros, ddof=1)

5441.23147825931